### PR TITLE
Connect UI to api

### PR DESCRIPTION
**Pull request by JakobFaarGreg**
## Description
Added UI integration with deployed APIs.
Landing page fetches exercises from Exercises microservice.
Exercise page fetches an individual exercise from the Exercises microservice.
Verify queries, on Exercise page, runs a verification on the queries for this exercise against the uploaded model.

### Type of change
- [x] New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
Monkey tests

### Checklist:

- [x] You have added a clear title and description for the PR
- [x] The CI pass
- [x] There are no merge conflicts
- [x] The code satisfies the [code standards](https://docs.google.com/document/d/1R9n9MeuRwz3WSrnWZUjrbmGeosTlu1Xz8f5F7gu2AJE/edit#heading=h.e9xl4xfoldow) 
- [x] I have added docstrings for my code, where needed
- [x] Code implements intended functionality
- [x] All relevant tests pass (Black, White, Integration, Monkey, etc.)

### CODE REVIEW

**Review by fatimaosman99 — COMMENTED** [review]
Look fine. Few comments.

File: src/Pages/ExercisePage.tsx [hunk 1]
```diff
@@ -1,23 +1,68 @@
...
+  useEffect(() => {
+    if (!exerciseID) return;
+    fetch(`/api/v1/exercises/${exerciseID}`)
+      .then(async (res) => await res.json())
+      .then((data) => {
+        setExercise(data);
+      });
+  }, [exerciseID]);
+
+  const handleVerifyClick = async () => {
```
Comment (frejaz) [inline src/Pages/ExercisePage.tsx:34]: docstrings?

File: src/Pages/ExercisePage.tsx [hunk 2]
```diff
@@ -1,23 +1,68 @@
...
-      { query: 'B[] A[1] && B[2] --> C[3]' },
-    ],
+  const [exercise, setExercise] = useState<Exercise | undefined>();
+  const [verifierResult, setVerifierResult] = useState<IVerifierResult>();
+  const [queries, setQueries] = useState<Query[]>([]);
+  const [file, setFile] = useState<File>();
+
+  const { exerciseID } = useParams();
+
+  useEffect(() => {
```
Comment (frejaz) [inline src/Pages/ExercisePage.tsx:25]: docstrings?

**Review by frejaz — CHANGES_REQUESTED** [review]
Maybe add some comments

**Review by fatimaosman99 — CHANGES_REQUESTED** [review]
Looks good. Have two comments.

**Review by fatimaosman99 — APPROVED** [review]

**Review by frejaz — APPROVED** [review]
yes

**Pull request merged**